### ISSUE TITLE
BUG/ENH: poisson deviance

### ISSUE THREAD

**Issue by lorentzenchr** [statsmodels/statsmodels#3355]
https://github.com/statsmodels/statsmodels/blob/master/statsmodels/genmod/families/family.py#L372
Should the deviance function of the poisson distribution have the term `- (endog - mu)` added inside the sum, cf. function `resid_dev`? If the model does not contain an intercept term `np.sum(endog - mu)` might not be zero.
```python
return 2 * np.sum(freq_weights * (endog * np.log(endog_mu) - (endog - mu))) / scale
```

**Comment by josef-pkt** [comment]
Yes, looks like a bug to me for the no-intercept case. (I'm adding corner-case label to signal that this is bug only affects "uncommon" cases.)

We don't use deviance in any optimization, so redundant calculation doesn't make much difference in performance.

**Comment by thequackdaddy** [comment]
I think I was the culprit who introduced this when I added `freq_weights`.

For my education, can one of you 2 explain this problem further? I was always under the impression that when you fit a poisson model `np.sum(endog - mu)` approaches zero. I'm aware this behavior is specific to Poisson (or maybe more specifically to the related variance function). 

Thanks!

**Comment by josef-pkt** [comment]
@thequackdaddy 

It is or can be nonzero if no constant and no implicit constant is included. 
It is a property of all GLM/LEF models, including OLS that the sum of residuals is zero *if* a constant is included.

However, we don't rule out the case that a user specifies a model without a constant. In the OLS case this would correspond to a regression through the zero point.

The match between sum(endog) and sum(fittedvalues) also occurs for subgroups with categorical dummy variables, e.g. in contingency tables.

**Comment by josef-pkt** [comment]
related when I looked at this recently for Logit
https://mail.python.org/pipermail/scikit-learn/2016-December/001010.html

When the constant is penalized, then it cannot force `sum(endog) = sum(fittedvalues)`.

**Comment by josef-pkt** [comment]
This is an old behavior, the term is missing also in 0.6, and not related to freq_weights addition
